### PR TITLE
Issue #11446: update treeWalkerTest to use verifyInlineConfigParser a…

### PR DESCRIPTION
**Pull request by mahfouz72**
part of #11446 :
Update TreeWalker test to discard verify and use verifyWithInlineConfigParser

### CODE REVIEW

**Comment by mahfouz72** [comment]
all the tests after the migration to the inlined config were successful. but I got these failures and the end of executing mvn verify command. can someone help me with this?
pr

**Comment by MANISH-K-07** [comment]
@mahfouz72 ,

PMD failures are related to priority violations..
Looks like your violations are not meeting the requirement here.

As you can see, the rule _JUnitTestsShouldIncludeAssert_ has a priority 3, so try modifying your violations accordingly ??

**Comment by nrmancuso** [comment]
@mahfouz72 please always share output as copy-pasted CLI output, like so:
```
[INFO] --- maven-pmd-plugin:3.21.2:check (default) @ checkstyle ---
[INFO] PMD version: 6.55.0
[INFO] PMD Failure: com.puppycrawl.tools.checkstyle.TreeWalkerTest:195 Rule:JUnitTestsShouldIncludeAssert Priority:3 JUnit tests should include assert() or fail().
[INFO] PMD Failure: com.puppycrawl.tools.checkstyle.TreeWalkerTest:234 Rule:JUnitTestsShouldIncludeAssert Priority:3 JUnit tests should include assert() or fail().
[INFO] PMD Failure: com.puppycrawl.tools.checkstyle.TreeWalkerTest:362 Rule:JUnitTestsShouldIncludeAssert Priority:3 JUnit tests should include assert() or fail().
[INFO] PMD Failure: com.puppycrawl.tools.checkstyle.TreeWalkerTest:422 Rule:JUnitTestsShouldIncludeAssert Priority:3 JUnit tests should include assert() or fail().
[INFO] PMD Failure: com.puppycrawl.tools.checkstyle.TreeWalkerTest:615 Rule:JUnitTestsShouldIncludeAssert Priority:3 JUnit tests should include assert() or fail().
[INFO] PMD Failure: com.puppycrawl.tools.checkstyle.TreeWalkerTest:633 Rule:JUnitTestsShouldIncludeAssert Priority:3 JUnit tests should include assert() or fail().
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  13:25 min
[INFO] Finished at: 2024-01-31T03:15:29Z

```

I will investigate this for you.

**Comment by nrmancuso** [comment]
> @mahfouz72 ,
> 
> PMD failures are related to priority violations.. Looks like your violations are not meeting the requirement here.
> 
> As you can see, the rule _JUnitTestsShouldIncludeAssert_ has a priority 3, so try modifying your violations accordingly ??

@MANISH-K-07 this comment is not very helpful, can you please explain exactly what a priority violation is?

**Comment by nrmancuso** [comment]
@mahfouz72 let's break this up into a few updates to get PRs merged faster. Please do:
1. Remove all updates to `execute` from this PR (keep only `verifyWithInlineConfigParser` conversions)
2. With the updates left in this PR, please remove file generation and create real input files. I will place a suggestion on one of your test methods to give you a clear idea of what to do.
3. We will approach remaining test methods in a separate PR (those that now use `execute`).

If this sounds good, please proceed. You will need to replace the suppression in `pom.xml` for now.

File: src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java [hunk 1]
```diff
@@ -93,18 +92,22 @@ protected String getPackageLocation() {
...
             writer.write(content);
         }
         final String[] expected1 = {
-            "1:45: " + getCheckMessage(ConstantNameCheck.class,
-                    MSG_INVALID_PATTERN, "k", "^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$"),
+            "10:27: " + getCheckMessage(ConstantNameCheck.class,
+                        MSG_INVALID_PATTERN, "k", "^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$"),
         };
-        verify(checkConfig, file.getPath(), expected1);
+        verifyWithInlineConfigParser(file.getPath(), expected1);
```
Comment (nrmancuso) [inline src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java:110]:
```suggestion
    @Test
    public void testProperFileExtension() throws Exception {
        final String path = getPath("InputProperFileExtension.java");
        final String[] expected = {
            "10:27: " + getCheckMessage(ConstantNameCheck.class,
                        MSG_INVALID_PATTERN, "k", "^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$"),
        };
        verifyWithInlineConfigParser(path, expected);
    }

```

Where `InputProperFileExtension.java` is in the folder specified in the `getPackageLocation()` method above, and looks like this:
```java
/*
* com.puppycrawl.tools.checkstyle.checks.naming.ConstantNameCheck
*/

package com.puppycrawl.tools.checkstyle.treewalker;

public class InputProperFileExtension {

    public static final int k = 5 + 4;
    // violation above
}


```

Comment (nrmancuso) [reply src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java:110]: This is an example of item (2) from https://github.com/checkstyle/checkstyle/pull/14388#issuecomment-1918347975.

**Comment by MANISH-K-07** [comment]
> @MANISH-K-07 this comment is not very helpful, can you please explain exactly what a priority violation is?

@nrmancuso 
I am not entirely sure if this is accurate but....
Each rule has a assigned priority (1 - high, 2 - medium-high, 3 - medium, 4 - medium-low, 5 - low). This priority is also used for the violation. In this case, the rule **_JUnitTestsShouldIncludeAssert_** has a priority 3.

Normally, if this rule wasn't failing, the lower priority (>=3) violations such as the ones we induce are expected to give off a warning rather than fail the build.
What I mean to say is few test methods have been modified by @mahfouz72  where there is no usage of assert() which comes under a higher priority violation hence the build failure.

File: src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java (hunk 1 above)
Comment (mahfouz72) [reply src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java:110]: AllTestsTest.testAllInputsHaveTest:60->walk:87->lambda$testAllInputsHaveTest$1:61->verifyInputFile:170->verifyInputFile:202 Resource must be named after a Test like 'InputMyCustomCase.java' and be in the sub-package of the test like 'mycustom' for test 'MyCustomCheckTest': com\puppycrawl\tools\checkstyle\treewalker\InputProperFileExtension.java
expected to be true

I got this error after making this change

Comment (nrmancuso) [reply src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java:110]: Try to rename file like `InputTreewalkerProperFileExtension`

Comment (mahfouz72) [reply src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java]: @nrmancuso Yes exactly. I renamed it and it worked

File: src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java [hunk 2]
```diff
@@ -131,16 +124,15 @@ public void testProperFileExtension() throws Exception {
...
-                    OneTopLevelClassCheck.MSG_KEY, "InputTreeWalkerInner"),
+            "10:1: " + getCheckMessage(OneTopLevelClassCheck.class,
+                        OneTopLevelClassCheck.MSG_KEY, "InputTreeWalkerInner"),
         };
         try (MockedConstruction<TreeWalkerAuditEvent> mocked =
-                 Mockito.mockConstruction(TreeWalkerAuditEvent.class, (mock, context) -> {
-                     throw new CheckstyleException("No audit events expected");
-                 })) {
-            verify(checkConfig, getPath("InputTreeWalker.java"), expected);
+                     Mockito.mockConstruction(TreeWalkerAuditEvent.class, (mock, context) -> {
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java:132]: Restore indentation as it was before

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java (hunk 2 above)
Comment (mahfouz72) [reply src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java]: @romani done. I should disable IntelliJ reformate on paste

**Review by nrmancuso — APPROVED** [review]
@mahfouz72 thanks a lot! This is a major improvement to these test cases, both from a readability and maintenance standpoint.

**Comment by nrmancuso** [comment]
@mahfouz72 just some advice: to help you organize things easily on your end, I suggest naming your branches like `issue-11446-treewalker`.

File: src/test/resources/com/puppycrawl/tools/checkstyle/treewalker/InputTreeWalkerSuppressionCommentFilter.java [hunk 3]
```diff
@@ -1,11 +1,19 @@
...
+
+com.puppycrawl.tools.checkstyle.filters.SuppressionCommentFilter
+checkCPP = false
+
+*/
 package com.puppycrawl.tools.checkstyle.treewalker;
 
 public class InputTreeWalkerSuppressionCommentFilter {
-    private int I;
+    private int I;    //violation
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/treewalker/InputTreeWalkerSuppressionCommentFilter.java:12]: Please add message part to violation message, as we do in other Inputs.
Please do this for all lines you updated in this PR

**Review by romani — CHANGES_REQUESTED** [review]
Last minor:

File: src/test/resources/com/puppycrawl/tools/checkstyle/treewalker/InputTreeWalkerSuppressionCommentFilter.java (hunk 3 above)
Comment (mahfouz72) [reply src/test/resources/com/puppycrawl/tools/checkstyle/treewalker/InputTreeWalkerSuppressionCommentFilter.java]: @romani done

**Review by romani — APPROVED** [review]
Ok to merge

**Comment by romani** [comment]
I got we are close to final fix

**Pull request merged**